plugin/pkg/auth/authenticator/password/keystone/keystone.go

```diff
@@ -39,11 +39,30 @@ type KeystoneAuthenticator struct {
 
 // AuthenticatePassword checks the username, password via keystone call
 func (keystoneAuthenticator *KeystoneAuthenticator) AuthenticatePassword(username string, password string) (user.Info, bool, error) {
-	opts := gophercloud.AuthOptions{
+	var opts gophercloud.AuthOptions
+	opts = gophercloud.AuthOptions{
 		IdentityEndpoint: keystoneAuthenticator.authURL,
 		Username:         username,
 		Password:         password,
 	}
+	if strings.HasSuffix(keystoneAuthenticator.authURL, "/v3") {
+		// username has a format of domain\name.
+		firstIndex := strings.Index(username, "\\")
```

Comment:
can usernames or domains contain a `\` character? if so, we can't assume this is a safe split character.